### ISSUE TITLE
Newest ha-bambulab update is breaking integration

### ISSUE THREAD

**Issue by klausiklau** [nberktumer/ha-bambu-lab-p1-spaghetti-detection#25]
Hello everyone, 

newest Dev update for ha-bambulab is changing the video feed from an image to a camera entity. 

so we can not choose the camera entity when using the blueprint: 

![Screenshot - 2024-10-22_12-48](https://github.com/user-attachments/assets/88f4dac1-1879-4595-a34f-25721e2148af)
(only available image entity shown is the cover image)

Thanks
Philipp

**Comment by klausiklau** [comment]
This is the official release now from ha-bambulab 
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.23

This will break spaghetti detection 
@nberktumer you might want to look into this?

Thanks
Philipp

**Comment by nberktumer** [comment]
I'll take a look into this, but the fix may take some time. In the mean time, if the old ha-bambulab v2.0.22 still works (I cannot check as I don't have access to my printer for the next couple of weeks), you can try downgrading the integration.

**Comment by jmdevita** [comment]
Albeit not a perfect solution, I created a pull request that should solve for the above issue. I couldn't find a way to isolate when a camera frame is updated, so instead I created a new blueprint variable "printer_frame" that users could link a sensor/image to as a trigger. For "old" versions ( <= 2.0.22 ), users should be able to just just map the image entity to this variable with no change. But for newer versions, after the breaking change (>= 2.0.23), users can map a Bambu Lab sensor that updates often. During testing, I mapped _P1S Current layer_, which worked great.
